### ISSUE TITLE
Android UiSettings.ZoomControls enabled not processed

### ISSUE THREAD

**Issue by professeurburp** [amay077/Xamarin.Forms.GoogleMaps#371]
### BUG / PROBREM REPORT
Android UiSettings.ZoomControlsEnabled not effective is set to false.

### SUMMARY

Even if UiSetting.ZoomControlsEnabled are set to false, + and - overlay controls still appear on Android platform.

### VERSIONS

* Xamarin.Forms.GoogleMaps - 2.1.0
* Xamarin.Forms - latest stable

### PLATFORMS

- [x] Android
- [ ] iOS
- [ ] UWP

### HOW TO REPRODUCE

Set Map.UiSettings.ZoomControlsEnabled = false on map in constructor, + and - zoom overlay controls still appear. Explanation is present in PR #370, but PR is closed since fix is not really trivial.

**Comment by professeurburp** [comment]
Actually, there is a side effect on deprecated HasZoomEnabled Map property and zoom UiSettings, in OnMapReady() Android renderer, where UiSettings are processed before deprecated properties. Since there is only a single HasZoomEnabled, that sets both ZoomControlsEnabled and ZoomGesturesEnabled native map uisettings, this property overrides previously set values from XF Map UiSettings.

Works perfectly if you set any zoom ui setting after map is ready, though.

**Comment by francedot** [comment]
@professeurburp Did you find a workaround meanwhile?

**Comment by professeurburp** [comment]
@francedot Sure, actually I think this is by design.

The thing is all map settings (UiSettings ones and Bindable ones) are processed in *OnMapReady()* function in Android *MapRenderer*. As for zoom controls, *UiSettings* are processed before zoom control BindableProperty:

```c#
void OnMapReady(GoogleMap map)
{
                ...
                _uiSettingsLogic.Initialize();
                ...
                UpdateHasZoomEnabled();
                ...
}
```

UiSettings are therefore overridden if you try and set them in your View Constructor, as I was doing, since *OnMapReady()* has not been called yet:

```c#
public LiveMapView()
{
            InitializeComponent();
            ...
            LiveMap.UiSettings.ZoomControlsEnabled = false;
            LiveMap.InitialCameraUpdate =
                 CameraUpdateFactory.NewPositionZoom(
                                new Position(48.858869, 2.233891), 15d);
}
```

For now, with current version, you have to set *UiSettings.ZoomControlsEnabled* after your View has been created, to ensure OnMapReady() has been called.

**Comment by amay077** [comment]
Sorry I am late.
I will try resolve this issue soon.

**Comment by francedot** [comment]
Yes, I confirm. Modifying _UISettings_ in _OnAppearing_ or after the Visual Tree has rendered is the only way to workaround the issue.

**Comment by amay077** [comment]
I fixed it and released 2.2.0-beta2 https://www.nuget.org/packages/Xamarin.Forms.GoogleMaps/2.2.0-beta2 .
Could you try?

**Comment by francedot** [comment]
@amay077 version 2.2.0-beta2 fixed the issue.
